discovery-infra/junit_log_parser.py

```diff
@@ -43,66 +44,111 @@ class LogEntry:
-    fail_case: List[TestCase] = list()
-    message = f"{entry.msg}\n{entry.error if entry.error else ''}"
-
-    if is_duplicate_entry(entry, message, fail_cases):
-        return []
-
-    test_case = TestCase(name=entry.func, classname=suite_name, category=suite_name, timestamp=entry.time)
-    test_case.failures.append(CaseFailure(message=message, output=message, type=entry.level))
-    fail_case.append(test_case)
-
-    if entry.level != "fatal":
-        # Add test case with the same name so it will be marked in PROW as flaky
-        flaky_test_case = TestCase(name=entry.func, classname=suite_name, category=suite_name)
-        fail_case.append(flaky_test_case)
-
-    return fail_case
-
-
-def get_failure_cases(log_file_name: Path, suite_name: str) -> List[TestCase]:
-    fail_cases: Dict[str, List[TestCase]] = dict()
-
-    with open(log_file_name) as f:
-        for line in f:
-            values = re.match(LOG_FORMAT, line)
-            if values is None:
-                continue
-
-            entry = LogEntry(**values.groupdict())
-            if entry.level not in EXPORTED_LOG_LEVELS:
-                continue
-            if entry.func not in fail_cases:
-                fail_cases[entry.func] = list()
-            fail_cases[entry.func] += get_log_entry_case(entry, fail_cases, suite_name)
-
-    log.info(f"Found {len(fail_cases)} failures on {suite_name} suite")
-    return [c for cases in fail_cases.values() for c in cases]
-
-
-def export_service_logs_to_junit_suites(source_dir: Path, report_dir: Path):
-    suites = list()
-    for file in source_dir.glob("k8s_assisted-service*.log"):
-        suite_name = Path(file).stem.replace("k8s_", "")
-        log.info(f"Creating test suite from {suite_name}.log")
-        test_cases = get_failure_cases(file, suite_name)
-        timestamp = test_cases[0].timestamp if test_cases else None
-        suites.append(TestSuite(name=suite_name, test_cases=test_cases, timestamp=timestamp))
-
-    log.info(f"Generating xml file for {len(suites)} suites")
-    xml_report = to_xml_report_string(suites)
-    with open(report_dir.joinpath(f"junit_log_parser_{str(uuid.uuid4())[:8]}.xml"), "w") as f:
-        log.info(f"Exporting {len(suites)} suites xml-report with {len(xml_report)} characters to {f.name}")
-        f.write(xml_report)
+class LogsConverter:
+
+    @staticmethod
+    def is_duplicate_entry(entry: LogEntry, entry_message: str, fail_cases: Dict[str, List[TestCase]]) -> bool:
+        for case in fail_cases.get(entry.func, []):
+            if case.failures and case.failures[0].message == entry_message:
+                return True
+        return False
+
+    @classmethod
+    def get_log_entry_case(cls, entry: LogEntry, fail_cases: Dict[str, List[TestCase]], suite_name: str) -> \
+            List[TestCase]:
+        fail_case: List[TestCase] = list()
+        message = f"{entry.msg}\n{entry.error if entry.error else ''}"
```

Comment:
```suggestion
        message = f"{entry.msg}\n{entry.error}"
```
if it's empty then it will have the same effect with this shortened syntax

Reply:
If entry.error is empty it's None by default and that will cause  `str(None) --> "None"`